org.OpenT2T.Sample.SuperPopular.Lamp/Vera Dimmable Light Bulb/js/thingTranslator.js

```diff
@@ -0,0 +1,174 @@
+            if (res.statusCode != 200)
+            {
+                console.log(res);
+                deferred.reject(res);
+            }
+            else
+            {
+                deferred.resolve(res);
+            }
+        });
+
+        res.on('error', (e) => {
+            console.log('problem with response:' + e.message);
+            deferred.reject(e);
+        });
+    });
+
+    req.end();
+
+    return deferred.promise;
+}
+
+// module exports, implementing the schema
+module.exports = {
+
+    // numeric int device id
+    deviceId : null,
+
+    // server used to connect to the vera hub
+    relayServer : null,
+
+    // session token for the relay server
+    relaySessionToken : null,
+
+    // serial number of the vera hub
+    pkDevice : null,
+
+    // data structure which represents the device targetted by this translator
+    device: null,
+
+    initDevice: function(dev) {
+        var deferred = q.defer();
+        this.device = dev;
+
+        if (typeof this.device != 'undefined') {
+            if (typeof (this.device.props) !== 'undefined') {
+                var props = JSON.parse(this.device.props);
+
+                if (typeof (props.relay_session_token) !== 'undefined') {
+                    this.relaySessionToken = props.relay_session_token;
+                } else {
+                    console.log('props.relay_session_token is undefined.');
+                    setImmediate(deferred.reject);
+                }
+
+                if (typeof (props.relay_server) !== 'undefined') {
+                    this.relayServer = props.relay_server;
+                } else {
+                    console.log('props.relay_server is undefined.');
+                    setImmediate(deferred.reject);
+                }
+
+                if (typeof (props.pk_device) !== 'undefined') {
+                    this.pkDevice = props.pk_device;
+                } else {
+                    console.log('props.pk_device is undefined.');
+                    setImmediate(deferred.reject);
+                }
+
+                if (typeof (props.id) !== 'undefined') {
+                    this.deviceId = props.id;
+                } else {
+                    console.log('props.id is undefined.');
+                    setImmediate(deferred.reject);
+                }
+
+                deferred.resolve();
+            } else {
+                console.log('props is undefined.');
+                setImmediate(deferred.reject);
+            }
+        } else {
+            console.log('device is undefined.');
+            setImmediate(deferred.reject);
+        }
+
+        console.log('Javascript initialized.');
+        logDeviceState(this.device);
+
+        return deferred.promise;
+    },
+
+    turnOn: function() {
+        console.log('[' + this.deviceId + '] turnOn called.');
+        return sendPowerStateCommandToDevice(100, this.deviceId, this.relayServer, this.relaySessionToken, this.pkDevice);
```

Comment:
Unfortunately, these `this.` references won't work when running in JXCore, at least with the current implementation. That's because the JXCore API for invoking a JS function doesn't have a way to specify the implicit `this` parameter. But probably we should fix that rather than having all the translators implement a workaround.